### ISSUE TITLE
OSStatus error: [-25243] Security error has occurred.

### ISSUE THREAD

**Issue by xspyhack** [kishikawakatsumi/UICKeyChainStore#105]
I am trying to write to the keychain in iOS with `accessGroup` while testing on my device, I am getting the error `OSStatus error: [-25243] Security error has occurred.`

```
UICKeyChainStore *store = [UICKeyChainStore keyChainStoreWithService:@"xspyhack" accessGroup:@"39V4U5AVXX.com.company.xxx"];
 store[@"Host"] = @"text"
```

**Comment by codesourse** [comment]
I meet it too and i storing in UICKeyChainStore  is lost;

OSStatus error: [-34018] Security error has occurred.
OSStatus error: [-34018] Security error has occurred.

**Comment by xinsight** [comment]
With Xcode 7.0 (iOS 8.4/9.0) `+[UICKeyChainStore setString:forKey:]` and `+[UICKeyChainStore stringforKey:]` fails to on get/set with a different error code:
`OSStatus error: [-25291] Security error has occurred.`

http://www.osstatus.com/search/results?platform=all&framework=all&search=-25291
http://www.osstatus.com/search/results?platform=all&framework=all&search=-25243
http://www.osstatus.com/search/results?platform=all&framework=all&search=-34801 // not found?

**Comment by susieyy** [comment]
> I meet it too and i storing in UICKeyChainStore is lost;
> 
> OSStatus error: [-34018] Security error has occurred.
> OSStatus error: [-34018] Security error has occurred.

The same thing has occurred.

With Xcode6.4.0 and iOS8.4.1.

**Comment by wilferrel** [comment]
I get the same thing any resolution for this?

**Comment by maknz** [comment]
From what I've read elsewhere it's an Apple bug which can occur when running the app with the Xcode debugger. A possible solution flying around is to reboot the device. I see this every now and again, pretty weird.

**Comment by cloudhm** [comment]
I run the project in my iphone5c, and meet the same problem.

**Comment by j-fritsch** [comment]
Check out Apple's latest response to this bug:
https://forums.developer.apple.com/thread/4743

They seem to feel confident that this issue has been resolved in iOS 9.3.

> Mar 22, 2016 3:28 AM

> OK, here’s the latest.  This is a complex problem with multiple possible causes:

> - Some instances of the problem are caused by incorrect app signing.  You can easily distinguish this case because the problem is 100% reproducible.

> - Some instances of the problem are caused by a bug in how iOS supports app development (r. 23,991,853).  Debugging this was complicated by the fact that another bug in the OS (r. 23,770,418) masked its effect, meaning the problem only cropped up when the device was under memory pressure.
> We believe these problems were resolved in iOS 9.3.

> - We suspect that there may be yet more causes of this problem.

> So, if you see this problem on a user device (one that hasn’t been talked to by Xcode) that’s running iOS 9.3 or later, please do file a bug report about it.  Try to include the device system log in your bug report (I realise that can be tricky when dealing with customer devices; one option is to ask the customer to install Apple Configurator, which lets them view the system log).  And if you do file a bug, please post your bug number, just for the record.

> On behalf of Apple I’d like to thank everyone for their efforts in helping to track down this rather horrid issue.

> Share and Enjoy
